creator-node/src/index.ts

```diff
@@ -127,14 +99,84 @@ const startApp = async () => {
   }
 
-  // Initialize services that do not require the server, but do not need to be awaited.
-  serviceRegistry.initServicesAsynchronously()
+  const numWorkers = clusterUtils.getNumWorkers()
+  logger.info(`Spawning ${numWorkers} processes to run the Express app...`)
+  const firstWorker = cluster.fork()
+  // Wait for the first worker to perform one-time init logic before spawning other workers
+  firstWorker.on('message', (msg) => {
+    if (msg?.cmd === 'initComplete') {
+      for (let i = 0; i < numWorkers - 1; i++) {
+        cluster.fork()
+      }
+    }
+  })
 
-  // Some Services cannot start until server is up. Start them now
-  // No need to await on this as this process can take a while and can run in the background
-  serviceRegistry.initServicesThatRequireServer(appInfo.app)
+  for (const worker of Object.values(cluster.workers || {})) {
+    worker?.on('message', (msg) => {
+      if (msg?.cmd === 'setSpecialWorkerId') {
+        clusterUtils.specialWorkerId = msg?.val
+      }
+    })
+  }
 
-  // when app terminates, close down any open DB connections gracefully
+  cluster.on('exit', (worker, code, signal) => {
+    logger.info(
+      `Worker process with pid=${worker.process.pid} died because ${
+        signal || code
+      }. Respawning...`
+    )
+    const newWorker = cluster.fork()
+    if (clusterUtils.specialWorkerId === worker.id) {
+      logger.info(
+        'The worker that died was the special worker. Setting a new special worker...'
+      )
+      clusterUtils.specialWorkerId = newWorker.id
+      for (const worker of Object.values(cluster.workers || {})) {
+        worker?.send({ cmd: 'setSpecialWorkerId', val: newWorker.id })
+      }
+    }
+  })
+}
+
+// Workers don't share memory, so each one is its own Express instance with its own version of objects like serviceRegistry
+const startAppForWorker = async () => {
+  /**
+   * Setting a different port is necessary for OpenResty to work. If OpenResty
+   * is enabled, have the app run on port 3000. Else, run on its configured port.
+   * @returns the port number to configure the Content Node app
+   */
+  const getPort = () => {
+    const contentCacheLayerEnabled = config.get('contentCacheLayerEnabled')
+
+    if (contentCacheLayerEnabled) {
+      return 3000
+    }
+
+    return config.get('port')
+  }
+
+  logger.info(
+    `Worker process with pid=${process.pid} and worker ID=${cluster.worker?.id} is running`
+  )
+
+  await verifyConfigAndDb()
+
+  const nodeMode = config.get('devMode') ? 'Dev Mode' : 'Production Mode'
+
+  await serviceRegistry.initServices()
+  logger.info(`Initialized services (Node running in ${nodeMode})`)
+  const appInfo = initializeApp(getPort(), serviceRegistry)
+  logger.info('Initialized app and server')
+
+  // Make the first worker wait for some services to be fully up before spinning up other workers
+  serviceRegistry.initServicesAsynchronously()
+  if (cluster.worker?.id === 1) {
```

Comment:
everything besides re-enquing jobs should only be done on worker ID 1 because it's part of one-time init. we don't need this logic to be duplicated by a new worker if worker ID 1 goes down

Reply:
yep makes sense after our discussion